source/dem/non_uniform_insertion.cc

```diff
@@ -151,50 +169,41 @@ NonUniformInsertion<dim>::assign_insertion_points(
           (dim == 3) ? this->number_of_particles_z_direction : 1;
         for (unsigned int k = 0; k < dim_nz; ++k)
```

Comment:
I'm not a fan of this idea of "recalculating all points" even if you insert a subsection of them. This is going to be very expensive for no reason. You can deduce the local k, j, i from the  index and avoid all of this mega looping.

Reply:
revised